src/inputs/Select/Select.js

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import React, { useMemo, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { Stack, Autocomplete, TextField } from '@mui/material';
+import makeStyles from '@mui/styles/makeStyles';
+import { TooltipInfo } from '../../misc';
+import { BasicInputPlaceholder } from '../BasicInputs/BasicInputPlaceholder';
+import { getCommonInputStyles } from '../style';
+
+const useStyles = makeStyles(getCommonInputStyles);
+
+export const Select = (props) => {
+  const classes = useStyles();
+  const { id, labels, tooltipText, value, options, disabled, onChange, isDirty } = props;
+  const getLabel = useCallback((key) => options?.find((el) => el.key === key)?.label, [options]);
+
+  const optionValue = useMemo(() => {
+    return { key: value, label: getLabel(value) };
+  }, [getLabel, value]);
+
+  if (disabled) {
+    const selectedValues = optionValue.label ?? labels.noValues;
+    return (
+      <BasicInputPlaceholder
+        id={`select-input-${id}`}
+        label={labels.label ?? id}
+        tooltipText={tooltipText}
+        value={selectedValues}
+      />
+    );
+  }
+
+  return (
+    <Stack
+      data-cy={`select-input-${id}`}
+      direction="row"
+      spacing={1}
+      alignItems="center"
+      className={isDirty ? classes.dirtyInput : isDirty === false ? classes.notDirtyInput : ''}
+    >
+      <Autocomplete
+        id={id}
+        options={options}
+        noOptionsText={labels.noOptions}
+        value={optionValue}
+        isOptionEqualToValue={(option, value) => option.key === value.key}
+        onChange={(event, newValue) => onChange(newValue?.key)}
+        getOptionLabel={(option) => option?.label ?? ''}
+        renderOption={(props, option) => (
+          <li data-testid={'option-' + option.key} {...props} key={option.key}>
+            {option.label}
+          </li>
+        )}
+        style={{ width: 500 }}
+        ListboxProps={{ 'data-cy': 'select-input-listbox' }}
+        renderInput={(params) => (
+          <TextField {...params} data-cy="scenario-select-input" placeholder={labels.label} label={labels.label} />
```

Comment:
If this component is a generic selector, we should probably remove the mention to "scenario" in `data-cy="scenario-select-input"`